mps-sync-plugin-lib/src/main/kotlin/org/modelix/mps/sync/transformation/mpsToModelix/initial/NodeSynchronizer.kt

```diff
@@ -63,12 +65,20 @@ class NodeSynchronizer(
 
             val mpsConcept = node.concept
             val cloudParentNode = branch.getNode(parentNodeId)
-            val cloudChildNode = cloudParentNode.addNewChild(childLink, -1, MPSConcept(mpsConcept))
 
-            // save the modelix ID and the SNode in the map
-            nodeMap.put(node, cloudChildNode.nodeIdAsLong())
-
-            synchronizeNodeToCloud(mpsConcept, node, cloudChildNode)
+            try {
+                // duplicate check
+                throwExceptionIfChildExists(cloudParentNode, childLink, node)
+                val cloudChildNode = cloudParentNode.addNewChild(childLink, -1, MPSConcept(mpsConcept))
+                nodeMap.put(node, cloudChildNode.nodeIdAsLong())
+
+                synchronizeNodeToCloud(mpsConcept, node, cloudChildNode)
+            } catch (ex: ChildNodeExists) {
```

Comment:
`synchronizeNodeToCloud` throws the exception, this catch the handles the case if a child (or grandchild) threw an exception. I do not see how we can prevent it in the parent if-statements. (We can go down in the hierarchy and check in advance, but that makes the traversal twice instead of once.)

Reply:
I see what you meant here. Apologizes for misusing `ChildNodeExists` here.

Fixed in f47b16274c70b62ac34d787aaf078b982c78632e